js/questionset.js

```diff
@@ -320,7 +320,8 @@ H5P.QuestionSet = function (options, contentId, contentData) {
   var $template = $(template.render(params));
 
   // Randomize questions only on instantiation
-  if (params.randomQuestions && contentData.previousState === undefined) {
+  const previousstateNotAvailable = (contentData.previousState === undefined || Object.keys(contentData.previousState).length === 0) ?? false;
```

Comment:
I don't understand when contentData.previousState is an empty object? I see that elsewhere in this file we are relying on testing whethter contentData.previousState is set or not. Also, I don't think you have understood the JavaScript nullish coalescing operator: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

Reply:
@fnoks 
When the content was created for the first time we get the empty object.
I am not so sure about this statement - "I see that elsewhere in this file we are relying on testing whethter contentData.previousState is set or not" - It would be great if you can explain it more.
Nullish Coalescing - I understand that the value is not null but with the breaket it produces the correct value to make this condition work. If you are aligned with other options we can switch back to it.

Reply:
I think we should understand why we get an empty object. We have e.g. this test: https://github.com/h5p/h5p-question-set/blob/bfd19ab60538eafc862a8bb529f0367984ce1a12/js/questionset.js#L158.

In which case will "?? false" be executed?